test/gnmi/devicestatetest.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gnmi
+
+import (
+	"context"
+	"github.com/onosproject/onos-test/pkg/onit/env"
+	"github.com/onosproject/onos-topo/api/device"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+// TestDeviceState tests that a device is connected and available.
+func (s *TestSuite) TestDeviceState(t *testing.T) {
+	simulator := env.NewSimulator().AddOrDie()
```

Comment:
@kuujo 
It looks like we don't kill the resources after each test. After one run, I observed some of the containers are still creating even the tests are finished. I assume the goal is to release the resources and shut down the cluster after the tests are terminated with success or failure.  Am I missing something?

```
onit run test -i onosproject/onos-config-tests:latest
‣ 2019-11-19T18:03:06-05:00 feasible-turtle Starting test
‣ 2019-11-19T18:03:06-05:00 feasible-turtle Deploy test coordinator
✓ 2019-11-19T18:03:06-05:00 feasible-turtle Deploy test coordinator
✓ 2019-11-19T18:03:08-05:00 feasible-turtle Starting test
‣ 2019-11-19T18:03:08-05:00 feasible-turtle Run test
‣ 2019-11-19T23:03:07Z feasible-turtle-cli Setup namespace
‣ 2019-11-19T23:03:07Z feasible-turtle-ha Setup namespace
‣ 2019-11-19T23:03:07Z feasible-turtle-gnmi Setup namespace
✓ 2019-11-19T23:03:07Z feasible-turtle-gnmi Setup namespace
‣ 2019-11-19T23:03:07Z feasible-turtle-gnmi Set up RBAC
✓ 2019-11-19T23:03:07Z feasible-turtle-cli Setup namespace
‣ 2019-11-19T23:03:07Z feasible-turtle-cli Set up RBAC
✓ 2019-11-19T23:03:07Z feasible-turtle-ha Setup namespace
‣ 2019-11-19T23:03:07Z feasible-turtle-ha Set up RBAC
✓ 2019-11-19T23:03:07Z feasible-turtle-cli Set up RBAC
✓ 2019-11-19T23:03:07Z feasible-turtle-ha Set up RBAC
✓ 2019-11-19T23:03:07Z feasible-turtle-gnmi Set up RBAC
=== RUN   ha
--- PASS: ha (0.00s)
PASS
=== RUN   cli
‣ 2019-11-19T23:03:09Z feasible-turtle-cli Setup Atomix controller
=== RUN   gnmi
‣ 2019-11-19T23:03:10Z feasible-turtle-gnmi Setup Atomix controller
✓ 2019-11-19T23:03:18Z feasible-turtle-gnmi Setup Atomix controller
‣ 2019-11-19T23:03:18Z feasible-turtle-gnmi Setup Raft partitions
‣ 2019-11-19T23:03:18Z feasible-turtle-gnmi Setup onos-topo
‣ 2019-11-19T23:03:18Z feasible-turtle-gnmi Setup onos-config
✓ 2019-11-19T23:03:19Z feasible-turtle-gnmi Setup Raft partitions
✓ 2019-11-19T23:03:22Z feasible-turtle-cli Setup Atomix controller
‣ 2019-11-19T23:03:22Z feasible-turtle-cli Setup onos-topo
‣ 2019-11-19T23:03:22Z feasible-turtle-cli Setup onos-config
‣ 2019-11-19T23:03:22Z feasible-turtle-cli Setup onos-cli service
‣ 2019-11-19T23:03:22Z feasible-turtle-cli Setup Raft partitions
✓ 2019-11-19T23:03:22Z feasible-turtle-cli Setup Raft partitions
✓ 2019-11-19T23:03:27Z feasible-turtle-cli Setup onos-cli service
✓ 2019-11-19T23:03:43Z feasible-turtle-cli Setup onos-topo
✓ 2019-11-19T23:03:44Z feasible-turtle-gnmi Setup onos-topo
✓ 2019-11-19T23:03:50Z feasible-turtle-cli Setup onos-config
=== RUN   cli/TestConfigGetCLI
‣ 2019-11-19T23:03:51Z feasible-turtle-cli Add simulator devicesim-ample-hound
✓ 2019-11-19T23:03:55Z feasible-turtle-gnmi Setup onos-config
=== RUN   gnmi/TestDeviceState
‣ 2019-11-19T23:03:55Z feasible-turtle-gnmi Add simulator devicesim-vast-marten
✓ 2019-11-19T23:04:06Z feasible-turtle-cli Add simulator devicesim-ample-hound
=== RUN   cli/TestConfigGetCLI/...ux-id=1]/state/address
=== PAUSE cli/TestConfigGetCLI/...ux-id=1]/state/address
=== RUN   cli/TestConfigGetCLI/...-id=1]/state/transport
=== PAUSE cli/TestConfigGetCLI/...-id=1]/state/transport
=== RUN   cli/TestConfigGetCLI/...x-id=1]/state/priority
=== PAUSE cli/TestConfigGetCLI/...x-id=1]/state/priority
=== RUN   cli/TestConfigGetCLI/...state/source-interface
=== PAUSE cli/TestConfigGetCLI/...state/source-interface
=== RUN   cli/TestConfigGetCLI/...n[aux-id=0]/state/port
=== PAUSE cli/TestConfigGetCLI/...n[aux-id=0]/state/port
=== RUN   cli/TestConfigGetCLI/...aux-id=0]/state/aux-id
=== PAUSE cli/TestConfigGetCLI/...aux-id=0]/state/aux-id
=== CONT  cli/TestConfigGetCLI/...ux-id=1]/state/address
=== CONT  cli/TestConfigGetCLI/...aux-id=0]/state/aux-id
=== CONT  cli/TestConfigGetCLI/...n[aux-id=0]/state/port
=== CONT  cli/TestConfigGetCLI/...state/source-interface
=== CONT  cli/TestConfigGetCLI/...x-id=1]/state/priority
=== CONT  cli/TestConfigGetCLI/...-id=1]/state/transport
=== RUN   cli/TestPluginsGetCLI
‣ 2019-11-19T23:04:06Z feasible-turtle-cli Add simulator devicesim-polite-caiman
✓ 2019-11-19T23:04:07Z feasible-turtle-gnmi Add simulator devicesim-vast-marten
=== RUN   gnmi/TestModels
‣ 2019-11-19T23:04:07Z feasible-turtle-gnmi Add simulator devicesim-up-marmoset
✓ 2019-11-19T23:04:16Z feasible-turtle-cli Add simulator devicesim-polite-caiman
=== RUN   cli/TestPluginsGetCLI/...fig-interfaces-stratum
=== PAUSE cli/TestPluginsGetCLI/...fig-interfaces-stratum
=== RUN   cli/TestPluginsGetCLI/test1
=== PAUSE cli/TestPluginsGetCLI/test1
=== RUN   cli/TestPluginsGetCLI/test1#01
=== PAUSE cli/TestPluginsGetCLI/test1#01
=== RUN   cli/TestPluginsGetCLI/openconfig-openflow
=== PAUSE cli/TestPluginsGetCLI/openconfig-openflow
=== RUN   cli/TestPluginsGetCLI/openconfig-if-ip
=== PAUSE cli/TestPluginsGetCLI/openconfig-if-ip
=== RUN   cli/TestPluginsGetCLI/...hercules-platform-node
=== PAUSE cli/TestPluginsGetCLI/...hercules-platform-node
=== CONT  cli/TestPluginsGetCLI/...fig-interfaces-stratum
=== CONT  cli/TestPluginsGetCLI/...hercules-platform-node
=== CONT  cli/TestPluginsGetCLI/openconfig-if-ip
=== CONT  cli/TestPluginsGetCLI/openconfig-openflow
=== CONT  cli/TestPluginsGetCLI/test1#01
=== CONT  cli/TestPluginsGetCLI/test1
--- PASS: cli (67.21s)
    --- PASS: cli/TestConfigGetCLI (15.80s)
        --- PASS: cli/TestConfigGetCLI/...ux-id=1]/state/address (0.00s)
        --- PASS: cli/TestConfigGetCLI/...aux-id=0]/state/aux-id (0.00s)
        --- PASS: cli/TestConfigGetCLI/...n[aux-id=0]/state/port (0.00s)
        --- PASS: cli/TestConfigGetCLI/...state/source-interface (0.00s)
        --- PASS: cli/TestConfigGetCLI/...x-id=1]/state/priority (0.00s)
        --- PASS: cli/TestConfigGetCLI/...-id=1]/state/transport (0.00s)
    --- PASS: cli/TestPluginsGetCLI (10.03s)
        --- PASS: cli/TestPluginsGetCLI/...fig-interfaces-stratum (0.00s)
        --- PASS: cli/TestPluginsGetCLI/...hercules-platform-node (0.00s)
        --- PASS: cli/TestPluginsGetCLI/openconfig-if-ip (0.00s)
        --- PASS: cli/TestPluginsGetCLI/openconfig-openflow (0.00s)
        --- PASS: cli/TestPluginsGetCLI/test1#01 (0.00s)
        --- PASS: cli/TestPluginsGetCLI/test1 (0.00s)
PASS
Test run failed test job is not complete
✓ 2019-11-19T18:04:19-05:00 feasible-turtle Run test


```